stock_available_to_promise_release/models/stock_move.py

```diff
@@ -0,0 +1,154 @@
+        return (
+            self.picking_code == "outgoing"
+            and not self.product_id.type == "consu"
+            and not self.location_id.should_bypass_reservation()
+        )
+
+    def _ordered_available_to_promise(self):
+        if not self._should_compute_ordered_available_to_promise():
+            return 0.
+        available = self.product_id.with_context(
+            location=self.warehouse_id.lot_stock_id.id
+        ).virtual_available
+        return max(
+            min(available - self._previous_promised_qty(), self.product_qty),
+            0.,
+        )
+
+    def _previous_promised_quantity_domain(self):
+        domain = [
+            ("need_release", "=", True),
+            ("product_id", "=", self.product_id.id),
+            ("date_priority", "<=", self.date_priority),
+            ("warehouse_id", "=", self.warehouse_id.id),
+        ]
+        return domain
+
+    def _previous_promised_qty(self):
+        previous_moves = self.search(
+            expression.AND(
+                [
+                    self._previous_promised_quantity_domain(),
+                    [("id", "!=", self.id)],
+                ]
+            )
+        )
+        promised_qty = sum(
+            previous_moves.mapped(
+                lambda move: max(
+                    move.product_qty - move.reserved_availability, 0.
+                )
+            )
+        )
+        return promised_qty
+
+    @api.multi
+    def release_available_to_promise(self):
+        self._run_stock_rule()
+
+    def _prepare_move_split_vals(self, qty):
+        vals = super()._prepare_move_split_vals(qty)
+        # The method set procure_method as 'make_to_stock' by default on split,
+        # but we want to keep 'make_to_order' for chained moves when we split
+        # a partially available move in _run_stock_rule().
+        if self.env.context.get("release_available_to_promise"):
+            vals.update(
+                {"procure_method": self.procure_method, "need_release": True}
+            )
+        return vals
+
+    @api.multi
+    def _run_stock_rule(self):
+        """Launch procurement group run method with remaining quantity
+
+        As we only generate chained moves for the quantity available minus the
+        quantity promised to older moves, to delay the reservation at the
+        latest, we have to periodically retry to assign the remaining
+        quantities.
+        """
+        precision = self.env["decimal.precision"].precision_get(
+            "Product Unit of Measure"
+        )
+        for move in self:
+            if not move.need_release:
+                continue
+            if move.state not in ("confirmed", "waiting"):
+                continue
+            # do not use the computed field, because it will keep
+            # a value in cache that we cannot invalidate declaratively
+            available_quantity = move._ordered_available_to_promise()
```

Comment:
use the field "move._ordered_available_to_promise" instead

Reply:
There is the explanation in the comment just above why the field is not used ;)